src/js/components/ComboBox/Combobox.tsx

```diff
@@ -94,30 +98,42 @@ const ComboBox = ({
     setComboLabel(checked.length > 0 ? checked.reduce(createLabel, '') : buttonLabel);
     setFilterList(checked);
     setUncheckedFilterList(unchecked);
+    if (selectedFilters.length > 0) {
+      setHasFilters(true);
+    }
   };
 
-  React.useEffect(() => {
-    sortFilters();
-  }, [flyoutVisible]);
-
   const clearFilters = () => {
     setSelectedFilters([]);
     setHasFilters(false);
     onChange([]);
-    toggleFlyout(!flyoutVisible);
+    toggleFlyout(false);
   };
 
-  const applyFilters = () => {
+  const applyFilters = (triggerOnChange = true) => {
     if (selectedFilters.length > 0) {
       setComboLabel(filterList.reduce(createLabel, ''));
-      toggleFlyout(!flyoutVisible);
       setHasFilters(true);
-      onChange(selectedFilters);
+
+      if (triggerOnChange) {
+        toggleFlyout(false);
+        onChange(selectedFilters);
+      }
     } else {
       clearFilters();
     }
   };
 
+  React.useEffect(() => {
+    if (!!selected && (selected.length !== 0) && selected.some((item) => !(selectedFilters || []).includes(item))) {
```

Comment:
Is it possible to create a named function that returns this result? I think could be a good idea to make this piece of code more understandable